Improve labels on membership payment fields

Abban opened the pull request:
This replaces headings on the payment page with legends
making the form a little more compatible with screen
readers. 

Ticket: https://phabricator.wikimedia.org/T364950

moiikana commented:
the question for the banking information still seems to be in an old format / not adapted:
![Bildschirmfoto vom 2024-05-21 17-27-18](https://github.com/wmde/fundraising-app-frontend/assets/11670701/0822646b-9052-4915-8db3-8a5c0373cb47)

Abban commented:
The Bank Data field still isn't converted to a new field, and making those improvements will have to go through the PM. I added a ticket for it last week, which I just added a point to look at the labele to https://phabricator.wikimedia.org/T364953
